sktime/data/util.py

```diff
@@ -102,8 +114,41 @@ def timeshifted_split(inputs, lagtime: int, chunksize: int = 1000, stride: int =
             while t < len(data):
                 if t == len(data_lagged):
                     break
+                x = data[ix[t:min(t + chunksize, len(data))]]
                 if lagtime > 0:
-                    yield data[t:min(t+chunksize, len(data))], data_lagged[t:min(t+chunksize, len(data_lagged))]
+                    x_lagged = data_lagged[ix[t:min(t + chunksize, len(data_lagged))]]
+                    yield x, x_lagged
                 else:
-                    yield data[t:min(t+chunksize, len(data))]
+                    yield x
                 t += chunksize
+
+
+class TimeSeriesDataset(object):
+    r"""
+    High-level container for time-series data. This is intended to be used together with pytorch data tools, i.e.,
+    data loaders and other utilities.
+    """
+
+    def __init__(self, data, lagtime: int = 0):
+        r"""Creates a new time series data set.
+
+        Parameters
+        ----------
+        data : iterable of data
+            The data which is wrapped into a dataset
+        lagtime : int, default=0
```

Comment:
and if zero?

Reply:
if zero the timeshifted data is equal to the not timeshifted data and the generator yields only instantaneous chunks

Reply:
ok cool as above i meant more like the physical interpretation but maybe that's more for the notebook